js/src/setup-mc/setup-stepper/setup-accounts/index.js

```diff
@@ -108,8 +111,15 @@ const SetupAccounts = ( props ) => {
 
 	const isGoogleAccountDisabled = jetpack?.active !== 'yes';
 	const isContinueButtonDisabled =
-		googleAdsAccount?.status !== 'connected' ||
-		googleMCAccount?.status !== 'connected';
+		hasAccess !== true ||
+		! hasFinishedResolution ||
+		! hasGoogleAdsConnection ||
+		( googleMCAccount?.status !== 'connected' &&
+			! (
+				googleMCAccount?.status === 'incomplete' &&
+				googleMCAccount?.step === 'link_ads' &&
+				hasAccess === true
+			) );
```

Comment:
Your guess is correct. This logic was initially added because the account setup step for linking an Ads account and MC account can now be initiated from either account (whichever is connected second). On the server, when this process happens for one of the accounts, the setup step is marked as complete for the other as well (e.g., [this line](https://github.com/woocommerce/google-listings-and-ads/blob/update/2262-ads-accept-invite-flow/src/Ads/AccountService.php#L368)). However, the data for the initial account is not updated for the front-end, so we let the user continue if either account is marked as incomplete because it's stuck on that step (similar to how we treat the billing step for the Ads account).

The combined logic for these two accounts is getting difficult to follow, so I've updated it in [95298bc](https://github.com/woocommerce/google-listings-and-ads/pull/2270/commits/95298bce433a14421a3b3d65dfcd27e24499d558) so the conditions for each account is separated.